test/e2e/common/pods.go

```diff
@@ -831,4 +831,101 @@ var _ = framework.KubeDescribe("Pods", func() {
 		validatePodReadiness(false)
 
 	})
+
+	ginkgo.It("pod should be in running state after the graceful terimination delete call", func() {
+		value := strconv.Itoa(time.Now().Nanosecond())
+		pod := getPodObject(podClient, f, value)
+		selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": value}))
+		options := metav1.ListOptions{LabelSelector: selector.String()}
+		listCompleted := make(chan bool, 1)
+		lw := &cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.LabelSelector = selector.String()
+				podList, err := podClient.List(options)
+				if err == nil {
+					select {
+					case listCompleted <- true:
+						e2elog.Logf("observed the pod list")
+						return podList, err
+					default:
+						e2elog.Logf("channel blocked")
+					}
+				}
+				return podList, err
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.LabelSelector = selector.String()
+				return podClient.Watch(options)
+			},
+		}
+		_, _, w, _ := watchtools.NewIndexerInformerWatcher(lw, &v1.Pod{})
+		defer w.Stop()
+
+		ginkgo.By("submitting the pod to kubernetes")
+		podClient.CreateSync(pod)
+
+		// We need to wait for the pod to be running, otherwise the deletion
+		// may be carried out immediately rather than gracefully.
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+
+		ginkgo.By("verifying the pod is in kubernetes")
+		pods, err := podClient.List(options)
+		framework.ExpectNoError(err, "failed to query for pods")
+		gomega.Expect(len(pods.Items)).To(gomega.Equal(1))
+		pod = &pods.Items[0]
+
+		ginkgo.By("deleting the pod")
+		err = podClient.Delete(pod.Name, metav1.NewDeleteOptions(30))
+		framework.ExpectNoError(err, "failed to delete pod")
+
+		ginkgo.By("verifying pod deletion was observed")
+		deleted := false
+		var lastPod *v1.Pod
+		var deleteObservedTime time.Time
+		timer := time.After(framework.DefaultPodDeletionTimeout)
+		for !deleted {
+			select {
+			case event, _ := <-w.ResultChan():
+				switch event.Type {
+				case watch.Deleted:
+					deleteObservedTime = time.Now()
+					lastPod = event.Object.(*v1.Pod)
+					deleted = true
+				case watch.Error:
+					e2elog.Logf("received a watch error: %v", event.Object)
+					framework.Failf("watch closed with error")
+				}
+			case <-timer:
+				framework.Failf("timed out waiting for pod deletion")
+			}
+		}
+		if !deleted {
+			framework.Failf("Failed to observe pod deletion")
+		}
+		gomega.Expect(deleteObservedTime).Should(gomega.BeTemporally(">", lastPod.DeletionTimestamp.Time), "pod failed to run up to grace period seconds")
```

Comment:
Most of the aspects of this PR are not correct -
1. **E2E Name:** `pod should be in running state after the graceful terimination delete call`
- Pod may or may not be in running state after delete call. It's totally depend on the underlying process running and its response to SIGTERM. And till what time Pod is expected to be in running state after delete call? E2E name MUST be very expressive about behaviour.
2. 
```
gomega.Expect(deleteObservedTime).Should(gomega.BeTemporally(">", lastPod.DeletionTimestamp.Time)
```

Here, deleteObservedTime (_time when watch.Deleted event is observed i.e. object removal_) would always be **`greater than`** Pod deletionTimestamp (_time when delete call is made, took from event_). So, this would ALWAYS PASS.

3.
>What this PR does / why we need it:
> - added new e2e test case for the pod phase validation after the graceful termination of the pod and pod resource removal from the pods list of the namespace

Neither PodPhase is checked nor pod resource removal. 
I'm wondering, what exact podPhase is expected to be checked after graceful termination? There are chances that object info might not have consistent relevant data in watch. 2ndly Pod might be in running state till `watch.Deleted` event is observed (SIGKILL) and the object would removed after that .

@pontiyaraja, Could you please write E2E Description (~Conformance Doc) for better clarity about the behaviour to be tested. Code would be updatable accordingly.